File: python/acronym/acronym.py
def abbreviate(words: str) -> str:
    """
    Takes a string of words and returns their acronym.

    The acronym is made up of the first letter of each word.
    If a word starts with a non-alphabetic character, it is ignored.
    If a word is separated from the previous one by a hyphen, underscore or space,
    it is considered a new word and its first letter is included in the acronym.
    """
    return "".join([c.upper() for (i, c) in enumerate(words) if c.isalpha() and (i == 0 or words[i - 1] in ' -_')])

File: go/acronym/acronym.go
package acronym

import (
	"regexp"
	"strings"
)

// Abbreviate takes a string and returns its acronym.
//
// An acronym is a concatenation of the first letter of each word in
// the input string, with all letters capitalized.
//
// The function recognizes subsequent words by the presence of a space, hyphen, or
// underscore preceding the first letter of the word.
func Abbreviate(s string) string {
	r := regexp.MustCompile(`(^|\s|-|_)[a-zA-Z]`)
	matches := r.FindAllString(s, -1)

	res := ""
	for _, m := range matches {
		res += strings.ToUpper(strings.Trim(m, " -_"))
	}
	return res
}
